lib/rules/extends-correct-class.js

```diff
@@ -15,12 +15,36 @@ function getExtendsOption(node) {
   return null
 }
 
+/**
+ * Format the array of allowed super names for a given class
+ * into a string for reporting.
+ *
+ * @param {Array<string>} allowedSuperNames
+ * @example
+ * formatNames(['HTMLElement']) // 'HTMLElement'
+ * formatNames(['HTMLElement', 'CustomElement']) // '{HTMLElement, CustomElement}'
+ */
+function formatNames(allowedSuperNames) {
+  if (allowedSuperNames.length === 1) {
+    return allowedSuperNames[0]
+  }
+
+  return `{${allowedSuperNames.join(', ')}}`
+}
```

Comment:
Perhaps this could be:

```suggestion
const formatter = new Intl.ListFormat('en', { style: 'short', type: 'disjunction' })
function formatNames(allowedSuperNames) {
  return formatter.format(allowedSuperNames)
}
```